account_fiscal_position_partner_type/models/account_fiscal_position.py

```diff
@@ -0,0 +1,119 @@
+        domain_group = base_domain + [("country_group_id.country_ids", "=", country_id)]
+
+        # Build domain to search records with exact matching criteria
+        fpos = self.search(domain_country + state_domain + zip_domain, limit=1)
+        # return records that fit the most the criteria, and fallback on
+        # less specific fiscal positions if any can be found
+        if not fpos and state_id:
+            fpos = self.search(domain_country + null_state_dom + zip_domain, limit=1)
+        if not fpos and zipcode:
+            fpos = self.search(domain_country + state_domain + null_zip_dom, limit=1)
+        if not fpos and state_id and zipcode:
+            fpos = self.search(domain_country + null_state_dom + null_zip_dom, limit=1)
+
+        # fallback: country group with no state/zip range
+        if not fpos:
+            fpos = self.search(domain_group + null_state_dom + null_zip_dom, limit=1)
+
+        if not fpos:
+            # Fallback on catchall (no country, no group)
+            fpos = self.search(base_domain + null_country_dom, limit=1)
+        return fpos or False
+
+    @api.model
+    def get_fiscal_position(self, partner_id, delivery_id=None):
+        if not partner_id:
+            return self.env["account.fiscal.position"]
+        # This can be easily overridden to apply more complex fiscal rules
+        PartnerObj = self.env["res.partner"]
+        partner = PartnerObj.browse(partner_id)
+
+        # if no delivery use invoicing
+        if delivery_id:
+            delivery = PartnerObj.browse(delivery_id)
+        else:
+            delivery = partner
+        fp_id = self.env["account.fiscal.position"]
+        # Only type has been configured
+        if (
+            delivery.commercial_partner_id.fiscal_position_type
+            and not delivery.property_account_position_id
+            and not delivery.commercial_partner_id.property_account_position_id
+        ):
+            # First search only matching VAT positions
+            vat_required = bool(partner.vat)
+            fp = self._get_fpos_by_region_and_position_type(
+                delivery.country_id.id,
+                delivery.state_id.id,
+                delivery.zip,
+                vat_required,
+                delivery.commercial_partner_id.fiscal_position_type,
+            )
+            # Then if VAT required found no match, try positions that do not
+            # require it
+            if not fp and vat_required:
+                fp = self._get_fpos_by_region_and_position_type(
+                    delivery.country_id.id,
+                    delivery.state_id.id,
+                    delivery.zip,
+                    False,
+                    delivery.commercial_partner_id.fiscal_position_type,
+                )
+            if fp:
+                fp_id = fp.id
+        else:
+            fp_id = super(AccountFiscalPosition, self).get_fiscal_position(
+                partner_id, delivery_id
```

Comment:
Don't change keyword arguments:

```suggestion
                partner_id, delivery_id=delivery_id
```